cfe_internal/enterprise/federation/federation.cf

```diff
@@ -54,20 +54,34 @@ bundle agent config
         expression => strcmp("superhub", "$(data[role])"),
         scope => "namespace";
 
-      "am_transporter"
+      "am_pusher"
         and => {strcmp("superhub", "$(data[remote_hubs][$(remotes)][role])"),
                 strcmp("enabled", "$(data[remote_hubs][$(remotes)][target_state])"),
                 strcmp("push_over_rsync", "$(data[remote_hubs][$(remotes)][transport][mode])")},
-        comment => "Has an enabled remote superhub with push as transport method, should run transport ",
+        comment => "Has an enabled remote superhub with push as transport method, should run push transport",
+        scope => "namespace";
+
+      "am_puller"
+        and => {"am_superhub",
+                strcmp("enabled", "$(data[remote_hubs][$(remotes)][target_state])"),
+                strcmp("pull_over_rsync", "$(data[remote_hubs][$(remotes)][transport][mode])")},
+        comment => "Superhub with some enabled remote hub with pull as transport method, should run pull transport",
         scope => "namespace";
 
+      "am_transporter"
+        or => {"am_pusher", "am_puller"};
```

Comment:
am_pusher and am_transporter are both namespace scoped classes that should be accessible from other bundles, shouldn't am_transporter be namespace scoped also?

Reply:
Yes, just like it was before my change. Good catch!